Update method for SubjectsController

brarrow opened the pull request:
#355 

- [x]  Update method

- [x] Tests for update method

brarrow committed:
Added Update method in SubjectController

brarrow committed:
Added test for SubjectUpdate method

brarrow commented:
I have a problem.
testUpdateSuccess working incorrectly. I don't know why.  It returns 400 code.

malinink commented on src/main/java/easytests/api/v1/controllers/SubjectsController.java:
```diff
@@ -21,7 +24,7 @@
  * @author VeronikaRevjakina
  */
 @RestController("SubjectsControllerV1")
-@SuppressWarnings("checkstyle:MultipleStringLiterals")
```
Why don't you use both warnings suppress?

brarrow replied:
done

malinink commented on src/main/java/easytests/api/v1/controllers/SubjectsController.java:
```diff
@@ -60,13 +63,40 @@
+
+        final SubjectModelInterface subjectModel = this.getSubjectModel(subject.getId());
+
+        if (!this.acl.hasAccess(subjectModel.getUser())) {
```
use subject direct.

brarrow replied:
done

malinink commented on src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java:
```diff
@@ -133,9 +133,85 @@ public void testListForbidden() throws Exception {
+                .andExpect(content().string(""))
+                .andReturn();
+
+        verify(this.subjectsService, times(1)).save(subjectModel);
```
nope, we should verify, that subject model have been passed into save method have fields that we pass from request. i.e. newSubject newDescription.

brarrow replied:
done

malinink commented on src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java:
```diff
@@ -133,9 +133,85 @@ public void testListForbidden() throws Exception {
+    }
+
+    @Test
+    public void testUpdateIdFailed() throws Exception {
```
What do we test here?

brarrow replied:
test deleted

brarrow committed:
Merge branch 'dev' of https://github.com/greenbarrow/easy-tests into feature-355

# Conflicts:
#	src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java

brarrow committed:
Merge branch 'dev' of https://github.com/greenbarrow/easy-tests into feature-355

# Conflicts:
#	src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java

brarrow committed:
over test deleted

brarrow committed:
fixes

malinink commented on src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java:
```diff
@@ -133,9 +133,85 @@ public void testListForbidden() throws Exception {
+                .andExpect(content().string(""))
+                .andReturn();
+
+        verify(this.subjectsService, times(1)).save(subjectModel);
```
use captor

brarrow replied:
done

malinink commented on src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java:
```diff
@@ -133,9 +133,85 @@ public void testListForbidden() throws Exception {
-     */
+    @Test
+    public void testUpdateSuccess() throws Exception {
+        final SubjectModelInterface subjectModel = subjectsSupport.getModelFixtureMock(0);
```
real model needed

brarrow replied:
done

malinink commented on src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java:
```diff
@@ -133,9 +133,85 @@ public void testListForbidden() throws Exception {
+        mvc.perform(put("/v1/subjects")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(new JsonSupport()
+                        .with(id, 1)
```
really you need to take that fields from additionalModelMock(1)

brarrow replied:
done

malinink commented on src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java:
```diff
@@ -133,9 +133,85 @@ public void testListForbidden() throws Exception {
+    @Test
+    public void testUpdateNotFound() throws Exception {
+        when(this.acl.hasAccess(any(UserModelInterface.class))).thenReturn(true);
+
```
mock getSubject to be null

brarrow replied:
done

brarrow committed:
fixes

brarrow committed:
changes

brarrow committed:
buf

brarrow committed:
Merge branch 'dev' of https://github.com/greenbarrow/easy-tests into feature-355

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/9c43a885-851c-44ca-a240-71992ec89cec) was created.

brarrow committed:
final version

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/9c43a885-851c-44ca-a240-71992ec89cec) completed: **3 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/7d179117-99ae-403b-b3ee-b4716d018a50) was created.

greenbarrow commented:
Coveralls failed because i use @VeronikaRevjakina mapper from #358.

greenbarrow commented:
@malinink Konstantin Aleksandrovich, check this PR, please.

brarrow committed:
Merge pull request #8 from malinink/dev

update

brarrow committed:
deleted extra lines

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/ffe4b340-dd9c-4aa4-b9b4-2ba3fc7c20cc) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/ffe4b340-dd9c-4aa4-b9b4-2ba3fc7c20cc) completed: **4 updated code elements**

brarrow commented:
@malinink Coverage fails not because of my files :acl and userscontroller. Check,please.

malinink commented on src/main/java/easytests/api/v1/controllers/SubjectsController.java:
```diff
@@ -61,12 +66,18 @@
+    @PutMapping("")
+    public void update(@RequestBody Subject subject) throws UnidentifiedModelException, NotFoundException, ForbiddenException {
+
+        final SubjectModelInterface subjectModel = this.getSubjectModel(subject.getId());
```
вначале проверьте что есть getId, т.к. id не обязательное. Если null - UnidentifiedEXception.
тесты профиксите.

malinink replied:
а дык, все есть в checkUser, тогда получение модели сделайте после проверки.

malinink replied:
заберите последний дев, разрезолвите конфликты

malinink committed:
cs TODO

brarrow committed:
Merge pull request #9 from malinink/dev

update

brarrow committed:
some little fixes

brarrow committed:
little fixes
Merge branch 'dev'
# Conflicts:
#	src/main/java/easytests/api/v1/controllers/SubjectsController.java
#	src/main/java/easytests/api/v1/mappers/SubjectsMapper.java
#	src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java

brarrow committed:
Merge remote-tracking branch 'origin/feature-355' into feature-355

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/c042135e-c239-4d85-a0cc-76200be00110) was created.

brarrow committed:
final

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/1ff8ab6b-0b38-4531-8e6d-f2921c95fd72) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/c042135e-c239-4d85-a0cc-76200be00110) completed: **1 updated code elements**

brarrow committed:
checkstyle

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/1ff8ab6b-0b38-4531-8e6d-f2921c95fd72) completed: **1 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/5001dcbd-1ab5-4559-a4f3-6211943c8c0b) was created.

brarrow commented:
@malinink Konstantin Aleksandrovich, merge this PR, please.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/5001dcbd-1ab5-4559-a4f3-6211943c8c0b) completed: **1 updated code elements**

malinink commented on src/main/java/easytests/api/v1/controllers/SubjectsController.java:
```diff
@@ -84,25 +82,15 @@ public Identity create(@RequestBody Subject subject) throws BadRequestException,
-        }
+    @PutMapping("")
+    public void update(@RequestBody Subject subject) throws NotFoundException, BadRequestException {
+        final SubjectModelInterface subjectModel = this.getSubjectModel(subject.getId());
```
выходит так, что id не обязательное поле по спеке, а потому может быть null, предлагаю добавить проверку на наличие значения вначале.

brarrow replied:
замечательное решение! Ура! Наконец-то я с этим покончил! Спасибо!

brarrow committed:
last change

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/0f27b060-c1c9-4384-8378-cf57df13e6da) was created.

brarrow committed:
srry i was too exited.
fix

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/b6e8f605-6834-4eea-ab60-a04619cd4b70) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/0f27b060-c1c9-4384-8378-cf57df13e6da) completed: **1 updated code elements**

malinink commented:
@greenbarrow  у нас для этого есть специальный класс ошибок - Uninidentified, используйте его

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/b6e8f605-6834-4eea-ab60-a04619cd4b70) completed: **1 updated code elements**

brarrow committed:
check the case when we doesn't have subjectId
And test for it

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/618c8a34-9950-48c5-a837-d005729b83c5) was created.

brarrow commented:
@malinink Konstantin Aleksandrovich, merge this PR, please.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/618c8a34-9950-48c5-a837-d005729b83c5) completed: **1 updated code elements**

brarrow committed:
dummy

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/32831bde-6631-4698-827e-933df1801d57) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/32831bde-6631-4698-827e-933df1801d57) completed: **1 updated code elements**

malinink commented on src/main/java/easytests/api/v1/controllers/SubjectsController.java:
```diff
@@ -71,7 +69,7 @@
     public Identity create(@RequestBody Subject subject) throws BadRequestException, ForbiddenException {
         if (subject.getId() != null) {
-            throw new IdentifiedModelException();
+            throw new UnidentifiedModelException();
```
перепутали понятия, UnidentifiedModelException - это когда id должен быть, а его нет.
У вас теперь наоборот.

malinink commented on src/main/java/easytests/api/v1/controllers/SubjectsController.java:
```diff
@@ -84,24 +82,18 @@ public Identity create(@RequestBody Subject subject) throws BadRequestException,
+    @PutMapping("")
+    public void update(@RequestBody Subject subject) throws NotFoundException, BadRequestException {
+        if (subject.getId() == null) {
+            throw new IdentifiedModelException();
```
Unidentified

malinink commented on src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java:
```diff
@@ -23,6 +24,8 @@
 import org.junit.runner.RunWith;
 import org.mockito.ArgumentCaptor;
 import static org.mockito.BDDMockito.*;
+
```
уберите лишние переводы строк

malinink replied:
Также просьба подтянуть dev в разрезолвить конфликты

brarrow committed:
Merge pull request #10 from malinink/dev

upd

brarrow committed:
ltl fixes
Merge branch 'dev'

# Conflicts:
#	src/main/java/easytests/api/v1/controllers/SubjectsController.java
#	src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/fa24021a-7ffa-43fe-99b8-2dd68274880c) was created.

brarrow committed:
un

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/7b93182f-fdaa-4697-82cb-0f4101de450d) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/fa24021a-7ffa-43fe-99b8-2dd68274880c) completed: **1 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/malinink/easy-tests/inspections/7b93182f-fdaa-4697-82cb-0f4101de450d) completed: **1 updated code elements**
